### PR TITLE
Added more fields to Akismet

### PR DESCRIPTION
**Pull request by yifanlu**
Added more Akismet options in addition, the user-agent and comment_type will be sent to akismet.

``` yaml
akismet:
  type: comment
  author: name #field
  content: message #field
  email: email #field
  url: url #field
```

Note I replaced the "enabled" option with "type" for comment_type.

### CODE REVIEW

**Comment by yifanlu** [comment]
Just an update. Before this commit, akismet caught about 20/50 spam comments in the span of three days. After this update, akismet caught about 50/50 spam comments in the same span.

**Comment by eduardoboucas** [comment]
This looks great, thanks!

To be honest, I don't have a lot of experience using Akismet and its implementation on Staticman has been a bit of an experimentation, so I really appreciate your help on this. At the moment, the public instance is using my Akismet key, which isn't sustainable in the long-term because it comes with usage limits.

Ideally, we'd allow site admins to supply their own Akismet keys, which is a bit tricky for sites in open repositories, because that information would be open to the public eye. Do you have any thoughts on this?

(Sorry I haven't replied to the e-mail you sent me a few days ago. I was planning on replying when I had the time to properly look into the whole Akismet implementation.)

**Comment by yifanlu** [comment]
Yeah I can't think of a good way to do it. One way is to get people to encrypt their information with a public key you generate for your instance of Staticman. But that would introduce a whole layer of complexity and people can't easily switch between Staticman hosts.

File: lib/Staticman.js [hunk 1]
```diff
@@ -46,7 +46,7 @@ Staticman.prototype._applyTransforms = function (data) {
 }
 
 Staticman.prototype._checkForSpam = function (fields) {
-  if (!this.config.akismet || !this.config.akismet.enabled) return Promise.resolve(fields)
+  if (!this.config.akismet || !this.config.akismet.type) return Promise.resolve(fields)
```
Comment (eduardoboucas) [inline lib/Staticman.js:49]: (Sorry, finally having enough time to properly look into this!)

Why did you change this property?

Comment (yifanlu) [reply lib/Staticman.js:49]: No technical reason; I just thought it was a bit weird that this is the only property with a separate enable field. Other options are "enabled" when there's an non-empty field.

Comment (eduardoboucas) [reply lib/Staticman.js:49]: Sounds good to me!

**Comment by eduardoboucas** [comment]
Merging. Thanks! 🎉
